### PR TITLE
[SWP-5969] - Allow missing credentials to make use of injected IAM role credentials

### PR DESCRIPTION
**Pull request by jimmy718**
This is to support the migration from IAM users to IAM roles for applications running earlier versions of Laravel (< 8).

### CODE REVIEW

File: src/EventServiceProvider.php [hunk 1]
```diff
@@ -112,10 +110,22 @@ protected function createEventBridgeDriver(array $config): \Illuminate\Contracts
...
             $config['credentials'] = Arr::only($config, ['key', 'secret', 'token']);
         }
 
         return $config;
     }
+
+    /**
+     * Make sure some AWS credentials were provided to the configuration array.
+     *
+     * @return bool
```
Comment (rickschubert) [inline src/EventServiceProvider.php:123]: In this JavaDoc string, can you type the input argument as well?

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -60,7 +60,7 @@ jobs:
           run: |
             composer require "laravel/framework:${{ matrix.laravel }}" --no-interaction --no-update
             composer require "orchestra/testbench:${{ matrix.testbench }}" --no-interaction --no-update
-            composer update --prefer-dist --no-interaction --no-progress
+            composer update --prefer-dist --no-interaction --no-progress --no-plugins
```
Comment (rickschubert) [inline .github/workflows/run-tests.yml:63]: Why is this needed?

**Review by rickschubert — COMMENTED** [review]
This looks good! I will not request changes here but rather comment as PHP is not my field of expertise.

File: src/EventServiceProvider.php (hunk 1 above)
Comment (oaklees) [reply src/EventServiceProvider.php:123]: It's PHPDoc, which I guess is pretty much the same. We could indeed add `@param  array  $config` - for consistency, but believe we just took the code that was added as part of the [PR](https://github.com/Pod-Point/laravel-aws-pubsub/pull/64/files#diff-010b2120a082db1cebaf7d5ffa6e2d259fdad880cb9b6f350a262026cf2ffd15) to bring this functionality into the `v1` branch. 

Could we also bring in over the EventBridgeBroadcasterTest and SnsBroadcasterTest?

File: src/EventServiceProvider.php [hunk 3]
```diff
@@ -94,12 +92,12 @@ protected function registerEventBridgeBroadcaster()
      * @param  array  $config
      * @return \Illuminate\Contracts\Broadcasting\Broadcaster
      */
-    protected function createEventBridgeDriver(array $config): \Illuminate\Contracts\Broadcasting\Broadcaster
+    public function createEventBridgeDriver(array $config): \Illuminate\Contracts\Broadcasting\Broadcaster
```
[the remark this answers is not in the document]
Comment (oaklees) [reply src/EventServiceProvider.php:95]: It looks like this was to aid testing - a test was added as part of the [PR](https://github.com/Pod-Point/laravel-aws-pubsub/pull/64/files#diff-7f041742eb34ce3a8885ec00b01f91f3cf2cad705e6b794251d5a4edf406cb2f) for this functionality that creates the driver.

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (oaklees) [reply .github/workflows/run-tests.yml:63]: This is down to Composer 2 (the PHP dependancy management tool) prohibiting the automatic execution of library scripts unless explicitly allowed - they're like lifecycle hooks that are fired during dependancy installation. 

Can you see any reason this works on 0.4 but not 0.0.x @jimmy718?

**Review by oaklees — CHANGES_REQUESTED** [review]
Good stuff. Only thing for me is also bringing the tests over that cover the introduced functionality, and then also a little into look into why we now need `--no-scripts`.

**Review by oaklees — APPROVED** [review]
Great stuff 🚢 

Let's double check it works as expected with PP and then we can do the patch release 🚀
